airflow/hooks/hive_hooks.py

```diff
@@ -495,37 +497,49 @@ def get_metastore_client(self):
         """
         Returns a Hive thrift client.
         """
-        import hmsclient
         from thrift.transport import TSocket, TTransport
         from thrift.protocol import TBinaryProtocol
+        result = 1
         ms = self.metastore_conn
         auth_mechanism = ms.extra_dejson.get('authMechanism', 'NOSASL')
         if configuration.conf.get('core', 'security') == 'kerberos':
             auth_mechanism = ms.extra_dejson.get('authMechanism', 'GSSAPI')
             kerberos_service_name = ms.extra_dejson.get('kerberos_service_name', 'hive')
-
-        socket = TSocket.TSocket(ms.host, ms.port)
+        hosts_list = ms.host.split(';') #split by semicolon
+        for host_name in range(len(hosts_list)):
```

Comment:
I haven't got time to check the implementation/logic in details yet, but the variable `host_name` here is not making sense to me. It's index actually (from `range(len(...))`), but the name itself is misleading.

Alternatively, you can change this line into `for host_name in hosts_list:` AND change all `hosts_list[host_name]` below into `host_name`.